hw/ip_templates/otp_ctrl/dv/env/otp_ctrl_scoreboard.sv.tpl

```diff
@@ -906,6 +994,60 @@ else:
                   end
                 end
               end
+              DaiZeroize: begin
+                bit[TL_AW-1:0] otp_addr = get_scb_otp_addr();
+
+                `uvm_info(`gfn, $sformatf("Zeroizing Loc: Part Index :%s, PartZeroFieldAddr :%x",
+                                      part_idx.name, PART_OTP_ZEROIZED_ADDRS[part_idx]), UVM_LOW);
+                `uvm_info(`gfn, $sformatf("Zeroizing Loc: otp_addr :%x, dai_addr :%x",
+                                      otp_addr, dai_addr), UVM_LOW);
+                `uvm_info(`gfn, $sformatf("Zeroizing Loc: is_granule_64 :%x",
+                                      is_granule_64(dai_addr)), UVM_LOW);
+
+                // Need to check if Partition is zeroizable first. If not raise interrupt and set
+                // the status and error registers
+                if (!part_is_zeroizable(int'(part_idx))) begin
+                  predict_err(OtpDaiErrIdx, OtpAccessError);
+                end else begin
+                  bit [TL_DW*2-1:0] descrambled_val;
+                  bit [TL_DW-1:0]   rdata_ret_val = '{default:1};
+                  // Partition is zeroizable
+                  if (otp_addr == PART_OTP_ZEROIZED_ADDRS[part_idx]) begin
+                    `uvm_info(`gfn, $sformatf("%s is zeroized, Digest ignored in future checks",
+                                        part_idx.name), UVM_LOW);
+                    // Need to separate out individual partition digests and set the ignore flag.
+                    ignore_digest_zero_chk[part_idx.name] = 1;
+                  end
+
+                  predict_no_err(OtpDaiErrIdx);
+                  dai_wr_ip = 1;
+
+                  // TODO: OTP Array in the scoreboard contains descrambled data! Not an actual
+                  // reflection of the RTL. Need to figure out how to fix this
+                  descrambled_val = descramble_data('{default:1}, part_idx);
+
+                  otp_a[otp_addr] = (   is_secret(dai_addr) && !is_zeroized_addr(dai_addr)
+                                     && !is_digest(dai_addr))
+                                   ? descrambled_val[TL_DW-1:0]:'{default:1};
+                  rdata_ret_val = is_granule_64(dai_addr) ? $countones(rdata_ret_val)*2
```

Comment:
For granule 64 rdata_ret_val should be the addition of the count ones of each 32 bit chunks, not one of them times 2. This can give you trouble when you test zeroization with stuck-at-zero, meaning when some zeroize commands fail to set all bits to 1.